packages/angular_devkit/build_angular/src/builders/karma/index.ts

```diff
@@ -146,6 +146,14 @@ export function execute(
         logger: context.logger,
       };
 
+      // Workaround https://github.com/angular/angular-cli/issues/28271, by clearing context by default
+      // for single run executions.
+      if (karmaOptions.client?.clearContext === undefined) {
```

Comment:
We already set have a `singleRun` variable in the 99
https://github.com/angular/angular-cli/blob/e6c2c4f67b1ca5da546c3818eba035dd336a76a9/packages/angular_devkit/build_angular/src/builders/karma/index.ts#L99

Maybe since this is quite coupled with that option, move it under it?

Also you can simply this to
```ts
karmaOptions.client ??= {};
// leave Jasmine Spec Runner output visible in browser
karmaOptions.client.clearContext ??= singleRun ?? false;
```